src/index.js

```diff
@@ -43,14 +40,35 @@ export default function onClickOutsideHOC(WrappedComponent, config) {
       return ref.getInstance ? ref.getInstance() : ref;
     }
 
-    // this is given meaning in componentDidMount/componentDidUpdate
-    __outsideClickHandler = null;
+    __outsideClickHandler = event => {
+      if (typeof this.__clickOutsideHandlerProp === 'function') {
```

Comment:
does this need a typeof check? It feels like if this property exist, we can reasonably assume it's going to be a function. Given that this will now be checked continuously, making this as fast as it reasonably can be feels worth it:
```
__outsideClickHandler = event => {
  if (this.__clickOutsideHandlerProp) {
    return this.__clickOutsideHandlerProp(event);
  }

  ...
}
```

(with the same updates for the subsequent checks)

Reply:
Yeah, we need to only check for existance. I just have picked check against function as its explicit and probably not more costful than any other type of check (probably even better than smth like if (prop) as each coerced possibilities need to be checked)

Do u feel this needs to be changed? I assume this is a fast path as like ive mentioned it performs only a single check - instead guarding against nulls, undefineds, 0s, empty strings and whatnot

Reply:
What ive written still stands, although i now see that u have proposed a simpler (for human) check which needs to apply all coercion rules. Havent noticed it before as im answering from my phone, so my comment might have sounded a little bit disconnected from ur comment. Anyway - like ive said, i believe this is faster (function check) in modern engines, although ofc this is micro optimization anyway

Reply:
https://docs.google.com/presentation/d/1_eLlVzcj94_G4r9j9d_Lj5HRKFnq6jgpuPJtnmIBs88/edit#slide=id.g2134da681e_0_596

I wouldnt expect `typeof fn === 'function'` being much different from `obj !== undefined`, but can change to the latter